stdlib/builtins.pyi

```diff
@@ -403,28 +403,28 @@ class str(Sequence[str]):
     def __len__(self) -> int: ...
     def __lt__(self, x: str) -> bool: ...
     def __mod__(self, x: Any) -> str: ...
-    def __mul__(self, n: int) -> str: ...
+    def __mul__(self, n: SupportsIndex) -> str: ...
     def __ne__(self, x: object) -> bool: ...
     def __repr__(self) -> str: ...
-    def __rmul__(self, n: int) -> str: ...
+    def __rmul__(self, n: SupportsIndex) -> str: ...
     def __str__(self) -> str: ...
     def __getnewargs__(self) -> Tuple[str]: ...
 
 class bytes(ByteString):
     @overload
-    def __new__(cls: Type[_T], ints: Iterable[int]) -> _T: ...
+    def __new__(cls: Type[_T], ints: Iterable[SupportsIndex]) -> _T: ...
     @overload
     def __new__(cls: Type[_T], string: str, encoding: str, errors: str = ...) -> _T: ...
     @overload
-    def __new__(cls: Type[_T], length: int) -> _T: ...
+    def __new__(cls: Type[_T], length: SupportsIndex) -> _T: ...
     @overload
     def __new__(cls: Type[_T]) -> _T: ...
     @overload
     def __new__(cls: Type[_T], o: SupportsBytes) -> _T: ...
     def capitalize(self) -> bytes: ...
-    def center(self, __width: int, __fillchar: bytes = ...) -> bytes: ...
+    def center(self, __width: SupportsIndex, __fillchar: bytes = ...) -> bytes: ...
     def count(
-        self, __sub: Union[bytes, int], __start: Optional[SupportsIndex] = ..., __end: Optional[SupportsIndex] = ...
+        self, __sub: bytes | SupportsIndex, __start: SupportsIndex | None = ..., __end: SupportsIndex | None = ...
```

Comment:
To stay under the 120 character limit I either had to switch to a multi-line signature,
or replace `Optional` with the pipe operator.
I went with the latter is it generally looks a bit cleaner and results in a smaller diff.

Reply:
We now prefer `|` unions. Instead of worrying about "120 character limit", whatever that is, you should use black.